HAL.md

```diff
@@ -0,0 +1,110 @@
+# Hardware abstract layer(HAL) overview
+
+## Abstract
+Hardware abstract layer(HAL) defines a standard interface for hardware vendors to implement, which enables Tangle-Accelerator to be aware of lower-level driver implementations. HAL layer supports such as uart, GPIO, secure storage and other low-level operators inside endpoint. The endpoint is a daemon, offering monitor for the device and can send the message to tangle-accelerator. HAL layer lets the user only need to implement their device operators from abstract interface which defined inside `device.h` but not need to know how endpoint work.
```

Comment:
`can send` -> `being able to send`
`which defined` -> `defined` or `which is defined`
`how endpoint work` -> `how endpoint works`